src/python-common/ceph/deployment/drive_group.py

```diff
@@ -84,6 +84,9 @@ def from_json(cls, device_spec):
 
         return cls(**device_spec)
 
+    def to_json(self):
+        return self.__dict__.copy()
```

Comment:
```suggestion
        return {
            'paths': [p.name for p in self.paths],
            ...
```

Reply:
But don't we want to keep _everything_ in there? In the case where we want to dump a drivegroup spec from the specstore and load it back in

Reply:
Hm. 

1. `self.__dict__.copy()` doesn't work, cause `self.paths` is of type `List[Device]`. Nothing suitable for `json.dumps`
2. I'd be :+1: for returning the same json structure as we accept. I.e. if `from_json` expects `path` to be a `List[str]`, then `to_json` should also return a `List[str]` for `self.paths`

Reply:
So we'd end up with something similar to [this](https://github.com/ceph/ceph/blob/9310d56436d043ab604b417da9265e8b9fab79c1/src/python-common/ceph/deployment/service_spec.py#L362)

like:

``` python
    def to_json(self):
        # type: () -> Dict[str, Any]
        c = self.__dict__.copy()
        if self.paths:
            c['paths'] = [p.name for p in self.paths]
        return c
```